clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -71,18 +71,18 @@
     private static final String REPLICAS_KEY_NAME = "replicas";
     private static final String ISR_KEY_NAME = "isr";
 
-    private final Cluster cluster;
-    private final Map<String, Errors> errors;
+    private final Collection<Node> brokers;
+    private final List<TopicMetadata> topicMetadata;
 
-    /**
-     * Constructor for MetadataResponse where there are errors for some of the topics,
-     * error data take precedence over cluster information for particular topic
-     */
-    public MetadataResponse(Cluster cluster, Map<String, Errors> errors) {
+
+    public MetadataResponse(List<Node> brokers, List<TopicMetadata> topicMetadata) {
```

Comment:
Maybe? I could see users constructing their own metadata requests, but it seems less likely they'd be constructing responses (except perhaps in unit tests). It isn't difficult to adapt if we think it's an issue.